clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/AppEngineUtils.groovy

```diff
@@ -48,7 +48,10 @@ class AppEngineUtils {
       credentials.appengine.apps().services().versions().list(project, service.getId()).queue(batch, callback)
     }
 
-    batch.execute()
+    if (batch.size() > 0) {
```

Comment:
There's a similar function [here](https://github.com/spinnaker/clouddriver/blob/45ed523c0e7340b63016f0057d9bda7ede193800/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleCachingAgent.groovy#L64) to do this already.